imap_processing/hi/l1c/hi_l1c.py

```diff
@@ -8,22 +8,35 @@
 import numpy as np
 import pandas as pd
 import xarray as xr
+from numpy import typing as npt
 
 from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
 from imap_processing.cdf.utils import parse_filename_like
-from imap_processing.hi.l1a.science_direct_event import DE_CLOCK_TICK_S
+from imap_processing.hi.l1a.science_direct_event import (
+    DE_CLOCK_TICK_S,
+    HALF_CLOCK_TICK_S,
+)
 from imap_processing.hi.utils import (
     CoincidenceBitmap,
     create_dataset_variables,
     full_dataarray,
+    parse_sensor_number,
 )
 from imap_processing.spice.geometry import (
     SpiceFrame,
     frame_transform,
     frame_transform_az_el,
 )
+from imap_processing.spice.spin import (
+    get_instrument_spin_phase,
+    get_spin_data,
+)
 from imap_processing.spice.time import ttj2000ns_to_et
 
+SPIN_PHASE_BIN_CENTERS = np.arange(0.05, 360, 0.1) / 360
+N_SPIN_BINS = SPIN_PHASE_BIN_CENTERS.size
+SPIN_PHASE_BIN_EDGES = np.linspace(0, 360, N_SPIN_BINS + 1) / 360
```

Comment:
A few minor comments:

These could be subtly off using arange / linspace, see the warning note here: https://numpy.org/doc/2.1/reference/generated/numpy.arange.html#numpy.arange

I'd suggest calculating one from the other to make sure it is consistent.

```suggestion
N_SPIN_BINS = 3600
SPIN_PHASE_BIN_EDGES = np.linspace(0, 360, N_SPIN_BINS + 1) / 360
SPIN_PHASE_BIN_CENTERS = (SPIN_PHASE_BIN_EDGES[:-1] + SPIN_PHASE_BIN_EDGES[1:) / 2
```

Reply:
Cool, that seems obvious in hindsight.